server/lib/github.js

```diff
@@ -200,3 +202,64 @@ export async function checkGithubStars(githubHandle, accessToken) {
     }
   }
 }
+
+const githubUsernameRegex = new RegExp('[a-z\\d](?:[a-z\\d]|-(?=[a-z\\d])){0,38}', 'i');
+const githubRepositoryRegex = new RegExp('[a-z\\d](?:[a-z\\.\\d]|-(?=[a-z\\.\\d])){1,100}', 'i');
+const githubHandleRegex = new RegExp(`^${githubUsernameRegex.source}(/(${githubRepositoryRegex.source})?)?$`, 'i');
+const githubPathnameRegex = new RegExp(`^/${githubUsernameRegex.source}(/(${githubRepositoryRegex.source})?)?`, 'i');
+
+/**
+ * Return the github handle from a URL
+ *
+ * @param {string} url
+ * @returns {string|null} handle
+ *
+ * @example
+ * getGithubHandleFromUrl('https://github.com/opencollective/opencollective-frontend')
+ * => 'opencollective/opencollective-frontend'
+ */
+export const getGithubHandleFromUrl = url => {
+  try {
+    const { hostname, pathname } = new URL(url);
+    if (hostname !== 'github.com' || pathname.length < 2) {
+      return null;
+    }
+
+    const regexResult = pathname.match(githubPathnameRegex);
+    if (regexResult) {
+      const handle = trim(regexResult[0], '/');
+      if (githubHandleRegex.test(handle)) {
+        return handle;
+      }
+    }
+  } catch {
+    // Ignore invalid URLs
+  }
```

Comment:
@SudharakaP The [updated frontend](https://github.com/opencollective/opencollective-frontend/pull/7872) already resolves this by changing the field to an URL rather than a Github handle.

But I'll look at it since we want to prevent any regression with this PR. Thanks!